src/js/common/schemaform/helpers.js

```diff
@@ -303,7 +303,15 @@ export function getNonArraySchema(schema) {
     }
 
     if (newProperties !== schema.properties) {
-      return _.set('properties', newProperties, schema);
+      let newSchema = _.set('properties', newProperties, schema);
+      if (newSchema.required) {
+        const newRequired = _.intersection(Object.keys(newSchema.properties), newSchema.required);
+        if (newRequired.length !== newSchema.required.length) {
+          newSchema = _.set('required', newRequired, newSchema);
```

Comment:
So this will just let us update the page when reviewing, but will still fail validation when submitting?